### PR TITLE
Gateways for extensions

### PR DESCRIPTION
**Pull request by jsuchome**
See https://github.com/fuseml/extensions/pull/31 for example of new extension description, using these new features.

Related to https://github.com/fuseml/fuseml/issues/237

### CODE REVIEW

**Review by stefannica — APPROVED** [review]
Thanks, this looks great. A much needed improvement.

File: helpers/istio.go [hunk 1]
```diff
@@ -35,7 +35,9 @@ spec:
       protocol: HTTP
     hosts:
     - "{{ .Host }}"
-
+`
+	if serviceHost != "" {
+		templateSource += `
```
Comment (flaviodsr) [inline helpers/istio.go:40]: any reason to not include this on the template?

Comment (jsuchome) [reply helpers/istio.go:40]: Sorry, I do not understand the question...

Comment (flaviodsr) [reply helpers/istio.go:40]: I see you have a template and depending on the `serviceHost` value you concatenate it with another template. Why not have only one template that checks the value of `serviceHost` and include the "second template" conditionally? for example:

```
{{ if serviceHost != "" }}
second template content...
{{ end }}
```

IMO it makes it easier to read the template logic.

Comment (jsuchome) [reply helpers/istio.go:40]: ah, good point. I did not try to find out how to do it inside template so I did it the traditional way.
But I agree your solution is better

**Review by flaviodsr — APPROVED** [review]

**Pull request merged**